chore(k8s): new configuration format for manifest files (#6907)
garden-io/garden: docs/reference/action-types/Run/kubernetes-pod.md

[spec](#spec) > files

> POSIX-style paths to YAML files to load manifests from. Each file may contain multiple manifests.

Reviewer: `spec.files` also needs to be marked as deprecated in the joi schemas.

Author: Done in d2a5441906c5bb119eb36f79874d9d4af6ab2e35

| ------------------ | ------- | -------- |
| `array[posixPath]` | `[]`    | No       |

### `spec.manifestFiles[]`

[spec](#spec) > manifestFiles

> POSIX-style paths to YAML files to load manifests from. These files *can not* contain any Garden template strings. Each file can contain multiple manifests.

Reviewer: ```suggestion
POSIX-style paths to YAML files to load manifests from. Garden will *not* use the Garden Template Language to transform manifests in these files. Each file can contain multiple manifests.
```

| Type               | Default | Required |
| ------------------ | ------- | -------- |
| `array[posixPath]` | `[]`    | No       |

### `spec.manifestTemplates[]`

[spec](#spec) > manifestTemplates

> POSIX-style paths to YAML files to load manifests from. Each can contain multiple manifests, and can include any Garden template strings, which will be resolved before searching the manifests for the resource that contains the Pod spec for the Run.

Reviewer: Maybe the documentation can be improved something along these lines?

```suggestion
POSIX-style paths to YAML files to load manifests from. Each file may contain multiple manifests.

Garden will treat each manifestTemplate file as a template string expression, resolve it and then attempt to parse the resulting string as YAML.

Then it will find the resource matching the Pod spec for the Run ([See also `spec.resource`](#spec.resource)).
```

Author: Addressed this and the comment above in 20c1fa878e18d361cfe1e8dcc3352095344215ae.
